shortfin/python/shortfin_apps/llm/components/service.py

```diff
@@ -433,7 +433,36 @@ async def run(self):
                 fn,
                 "".join([f"\n  {i}: {ary.shape}" for i, ary in enumerate(args)]),
             )
-            # Invoke. Logits are of shape [bs, bsl, d].
+
+            # pre-invocation args dump
+            try:
```

Comment:
Kind of a nit, but I feel like a developer would expect `service.py` to handle checking the `SHORTFIN_DEBUG_LLM_SERVICE` var and invoking `pre_invocation_debug_dump` if True, instead of `debug_service.py` having control over that.

I think this may be a little cleaner if we made it a private `async def _pre_invocation_debug_dump` on `InferenceExecutorProcess`, initialized a constant `SHORTFIN_DEBUG_LLM_SERVICE` at the top of `service.py`, and then called:

```python
if SHORTFIN_DEBUG_LLM_SERVICE:
    await self._pre_invocation_debug_dump(...)
```

And moved the import & error handling logic to that private function, instead of `debug_service.py` reading the env variable and controlling whether or not it gets dumped with:

```python
if not SHORTFIN_DEBUG_LLM_SERVICE:
        return
```

Makes `run` a little less noisy too

Reply:
Yeah this is a bit strange, but I can see how injecting in this way is convenient.

This could also be configured via a new argument to the server, here: https://github.com/nod-ai/shark-ai/blob/main/shortfin/python/shortfin_apps/llm/server.py. That would be more discoverable than an environment variable, since it would appear in `--help` output.

Reply:
the intention is to make this attach to service.py at as few points as possible, so that if we want to make a more unified debug interface it's easy to do so. I do agree that this is a little cumbersome though.

Currently working on some simplifications, where instead of making the debug dump function a part of InferenceExecProcess, we pass self and all local variables to the dump function.